core/src/main/scala/codesearch/core/regex/space/SpaceInsensitive.scala

```diff
@@ -28,16 +28,20 @@ object SpaceInsensitive {
             current match {
               case SpecialSymbol("+") => current :: result
               case SpecialSymbol("*") => current :: result
-              case RepetitionSeq(_)   => current :: result
               case SpecialSymbol("?") => current :: (allocatedOneOrMoreSpaces ::: result.tail)
               case SpecialSymbol(" ") => current :: result
+              case RepetitionSeq(_)   => current :: result
               case _                  => current :: SpecialSymbol("+") :: result
             }
           case _ =>
             current :: result
         }
       }
-      .reverse
+      .reverse match {
+      case result @ _ :+ SpecialSymbol(" ") :+ SpecialSymbol(" ") => result
+      case result @ _ :+ SpecialSymbol(" ")                       => result :+ SpecialSymbol("+")
```

Comment:
It seems rather strange to me that this case has to be handled separately after the fold is done. Is there a way to rewrite the fold to avoid it?

cc @onewhl

Reply:
Well, yes, processing is only after foldLeft, it is true. I check the end of the finished sequence.

Reply:
Made - https://github.com/aelve/codesearch/pull/172/commits/af5f377acdada293020b9cc829da2f83c87523fb